src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2727,7 +2727,7 @@ object Types {
 
         case tp @ AnnotatedType(annot, underlying) =>
           val underlying1 = this(underlying)
-          if (underlying1 eq underlying) tp else underlying1
+          if (underlying1 eq underlying) tp else tp.derivedAnnotatedType(annot, underlying1)
```

Comment:
Good point. Yes, I think we should, so that annotations can refer to types.

Do we have any traverser that maps over all types occurring within a tree and all its subtrees?